scripts/tests.e2e.sh

```diff
@@ -30,15 +33,25 @@ trap cleanup EXIT
 echo "checking out target AvalancheGo version ${avalanche_version}"
 if [[ -d "${AVALANCHEGO_CLONE_PATH}" ]]; then
   echo "updating existing clone"
-  cd "${AVALANCHEGO_CLONE_PATH}"
-  git fetch
 else
   echo "creating new clone"
   git clone https://github.com/ava-labs/avalanchego.git "${AVALANCHEGO_CLONE_PATH}"
-  cd "${AVALANCHEGO_CLONE_PATH}"
 fi
-# Branch will be reset to $avalanche_version if it already exists
-git checkout -B "test-${avalanche_version}" "${avalanche_version}"
+
+cd "${AVALANCHEGO_CLONE_PATH}"
+git fetch origin ${avalanche_version}
+
+ref=""
+# check if given reference is a tag or branch
+if git rev-parse --quiet --verify "origin/${avalanche_version}^{commit}" >/dev/null; then
+  echo "checking out branch ${avalanche_version}"
+  ref="origin/${avalanche_version}"
+else
+  echo "checking out tag ${avalanche_version}"
+  ref="${avalanche_version}"
+fi
+
+git checkout -B "test-${avalanche_version}" "${ref}"
```

Comment:
Does it make sense for us to be targeting a branch for CI? It feels like we should be explicit and target either a tag or a commit imo.

Reply:
reverted it